src/main/java/frc/robot/subsystems/Arm/ArmSubsystem.java

```diff
@@ -0,0 +1,340 @@
+        }
+      } else {
+        this.stow();
+        if (piece == gamePiece.CUBE) {
+          finalState = ArmState.SHELF_CUBE;
+        } else {
+          finalState = ArmState.SHELF_CONE;
+        }
+        useFinal = true;
+      }
+    } else {logger.info("cannot pickup nonexistent game piece");}
+  }
+
+  public void floor(gamePiece piece, boolean isUpright) {
+    // if it's a cube, set isUpright to whatever! freedom!!!
+    if (piece != gamePiece.NONE) {
+      if (currState == ArmState.STOW) {
+        currState = ArmState.PARALLEL_TRANS;
+        if (piece == gamePiece.CUBE) {
+          desiredState = ArmState.FLOOR_CUBE;
+          shoulderSubsystem.setPos(ArmConstants.kCubeFloorShoulderPos);
+          extendoSubsystem.setPos(ArmConstants.kCubeFloorExtendoPos);
+          wristSubsystem.setPos(ArmConstants.kCubeFloorWristPos);
+        } else {
+          desiredState = ArmState.FLOOR_CONE;
+          if (isUpright) {
+            shoulderSubsystem.setPos(ArmConstants.kConeFloorUprightShoulderPos);
+            extendoSubsystem.setPos(ArmConstants.kConeFloorUprightExtendoPos);
+            wristSubsystem.setPos(ArmConstants.kConeFloorUprightWristPos);
+          } else {
+            shoulderSubsystem.setPos(ArmConstants.kConeFloorShoulderPos);
+            extendoSubsystem.setPos(ArmConstants.kConeFloorExtendoPos);
+            wristSubsystem.setPos(ArmConstants.kConeFloorWristPos);
+          }
+        }
+      } else {
+        this.stow();
+        if (piece == gamePiece.CUBE) {
+          finalState = ArmState.FLOOR_CUBE;
+        } else {
+          if (isUpright) {
+            finalState = ArmState.FLOOR_CONE_UPRIGHT;
+          } else {
+            finalState = ArmState.FLOOR_CONE;
+          }
+        }
+        useFinal = true;
+      }
+    } else {logger.info("cannot pickup nonexistent game piece");}
+  }
+
+  public void yoshi(gamePiece gamePiece) {
+    if (gamePiece == ) {
+      currState = ArmState.PARALLEL_TRANS;
+      desiredState = ArmState.YOSHIED;
+      extendoSubsystem.setPos(ArmConstants.kYoshiPos);
+      finalState = ArmState.FLOOR_CUBE;
+    }
+    if (currState == ArmState.FLOOR_CONE) {
+      currState = ArmState.PARALLEL_TRANS;
+      desiredState = ArmState.YOSHIED;
+      extendoSubsystem.setPos(ArmConstants.kYoshiPos);
+      finalState = ArmState.FLOOR_CONE;
+    }
+    if (currState == ArmState.FLOOR_CONE_UPRIGHT) {
+      currState = ArmState.PARALLEL_TRANS;
+      desiredState = ArmState.YOSHIED;
+      extendoSubsystem.setPos(ArmConstants.kYoshiPos);
+      finalState = ArmState.FLOOR_CONE_UPRIGHT;
+    }
+  }
+
+  public boolean isFinished() {
+    if (shoulderSubsystem.isFinished()
+        && extendoSubsystem.isFinished()
+        && wristSubsystem.isFinished()) {
+      return true;
+    }
+    return false;
+  }
+
+  @Override
+  public void periodic() {
+    if (this.isFinished()) {
+      currState = desiredState;
+      if (useFinal) {
```

Comment:
add a check to this to make sure currState is STOW (just in case useFinal doesn't get reset somewhere)